Quaver.Shared/IPC/QuaverIpcHandler.cs

```diff
@@ -54,7 +62,78 @@ public static void HandleProtocolMessage(string message)
                 if (game?.CurrentScreen is EditScreen screen)
                     screen.GoToObjects(message);
                 else
-                    NotificationManager.Show(NotificationLevel.Warning, "You must be in the editor to use this function!");
+                    NotificationManager.Show(NotificationLevel.Warning,
+                        "You must be in the editor to use this function!");
+            }
+            else if (message.StartsWith("map/"))
+            {
+                var game = (QuaverGame) GameBase.Game;
+                if (game.CurrentScreen.Type != QuaverScreenType.Select)
+                {
+                    NotificationManager.Show(NotificationLevel.Warning,
+                        "You must be in the song select screen to select a map!");
+                    return;
+                }
+
+                var screen = game.CurrentScreen as SelectionScreen;
+
+                message = message.Replace("map/", "");
+                int mapId;
+                if (!int.TryParse(message, out mapId))
+                {
+                    NotificationManager.Show(NotificationLevel.Warning, "The provided ID was not a number!");
+                    return;
+                }
+
+                // Check if we have the map installed
+                var map = MapManager.FindMapFromOnlineId(mapId);
+                if (map != null)
+                {
+                    MapManager.Selected.Value = map;
+
+                    lock (screen.AvailableMapsets.Value)
+                        screen.AvailableMapsets.Value = MapsetHelper.FilterMapsets(screen.CurrentSearchQuery);
+                    return;
+                }
+
+                if (!OnlineManager.Connected)
+                {
+                    NotificationManager.Show(NotificationLevel.Warning, "You must be logged in to download maps!");
+                    return;
+                }
+
+                // Check if the map exists online
+                var request = new APIRequestMapInformation(mapId);
+                var response = request.ExecuteRequest();
+                if (response.Status == 400 || response.Map?.MapsetId == -1)
+                {
+                    NotificationManager.Show(NotificationLevel.Warning, "The map does not exist!");
```

Comment:
Prolly "The map does not exist on the server!" or something like that, should be clearer that it's talking about quaver online maps